storage-proofs/benches/encode.rs

```diff
@@ -24,32 +56,36 @@ fn pregenerate_data<H: Hasher>(degree: usize) -> Pregenerated<H> {
     let data: Vec<u8> = (0..(degree + 1))
         .flat_map(|_| fr_into_bytes::<Bls12>(&rng.gen()))
         .collect();
-    let parents: Vec<usize> = (0..degree).map(|pos| pos).collect();
+
+    let graph = ZigZagBucketGraph::<H>::new_zigzag(degree + 1, degree, 8, new_seed());
     let replica_id: H::Domain = rng.gen();
+
     Pregenerated {
         data,
-        parents,
+        graph,
         replica_id,
     }
 }
 
-fn encode_single_node<H: Hasher>(
+fn encode_single_node<H: Hasher, G: Graph<H>>(
     data: &mut [u8],
-    parents: &[usize],
+    graph: &G,
     replica_id: &H::Domain,
     node: usize,
+    parents: &mut [usize],
 ) {
-    let key = vde::create_key::<H>(replica_id, node, parents, data).unwrap();
+    graph.parents(node, parents);
+    let key = vde::create_key::<H>(replica_id, node, &parents, data).unwrap();
     let start = data_at_node_offset(node);
     let end = start + NODE_SIZE;
 
     let node_data = H::Domain::try_from_bytes(&data[start..end]).unwrap();
-    let encoded = H::sloth_encode(&key, &node_data, 1);
+    let encoded = H::sloth_encode(&key, &node_data, 0);
     encoded.write_bytes(&mut data[start..end]).unwrap();
 }
 
 fn encode_single_node_benchmark(cc: &mut Criterion) {
-    let degrees = vec![3, 5, 10];
+    let degrees = vec![3, 5];
```

Comment:
Can we include degree 13, which is our current real value. That way we can get a direct measurement of the cost of encoding one real node.

Reply:
this is the base degree, and it uses an expansion degree of 8, so I believe we are already doing that

Reply:
I see. I missed the `8` hiding on line 60.